pkg/types/conversion/tfjson/tfjson.go

```diff
@@ -83,10 +83,19 @@ func tfJSONAttributeToV2Schema(attr *tfjson.SchemaAttribute) *schemav2.Schema {
 		Deprecated:  deprecatedMessage(attr.Deprecated),
 		Sensitive:   attr.Sensitive,
 	}
-	if err := schemaV2TypeFromCtyType(attr.AttributeType, v2sch); err != nil {
-		panic(err)
+	if attr.AttributeNestedType != nil {
+		if err := schemaV2TypeFromNestedType(attr.AttributeNestedType, v2sch); err != nil {
+			panic(err)
+		}
+		return v2sch
 	}
-	return v2sch
+	if !attr.AttributeType.Equals(cty.NilType) {
```

Comment:
Does `cty.NilType` mean that no attribute type has been set?

Reply:
Yes, we didn't hit such a case until recently but it looks like it can be nil and only have `AttributeNestedType`.